api/soap/mc_user_profile_api.php

```diff
@@ -71,3 +71,45 @@ function mc_user_profiles_get_all( $p_username, $p_password, $p_page_number, $p_
 		'results' => $t_paged_results
 	);
 }
+
+/**
+ * Sets the password for the user if possible
+ *
+ * @param string   $p_username    The user's username
+ * @param string   $p_password    The user's password
+ * @param string   $p_newpassword    The user's new password
+ *
+ */
+function mc_change_password( $p_username, $p_password , $p_newpassword  ){
+        $t_user_id = mci_check_login( $p_username, $p_password );
+        if( $t_user_id === false ) {
+                return mci_soap_fault_login_failed();
+        }
+        if( !helper_call_custom_function( 'auth_can_change_password', array() ) ){
```

Comment:
Use `auth_can_set_password()` instead of calling this helper method.  This will also do the check to make sure this specific user is not using a different kind of authentication.